src/main/java/org/opensearch/neuralsearch/search/collector/HybridCollapsingTopDocsCollector.java

```diff
@@ -0,0 +1,350 @@
+                    HybridQueryScorer hybridQueryScorer = getHybridQueryScorer(childScorable.child());
+                    if (Objects.nonNull(hybridQueryScorer)) {
+                        log.debug(
+                            String.format(
+                                Locale.ROOT,
+                                "found hybrid query scorer, it's child of scorer %s",
+                                childScorable.child().getClass().getSimpleName()
+                            )
+                        );
+                        return hybridQueryScorer;
+                    }
+                }
+                return null;
+            }
+
+            @Override
+            public void collect(int doc) throws IOException {
+                groupSelector.advanceTo(doc);
+                T groupValue = groupSelector.currentValue();
+                assert groupValue != null;
+                FieldValueHitQueue<FieldValueHitQueue.Entry>[] group = groupQueueMap.get(groupValue);
+                float[] subScoresByQuery = compoundQueryScorer.hybridScores();
+                if (group == null) {
+                    initializeQueue(subScoresByQuery.length);
+                }
+
+                if (initializeLeafComparatorsPerSegmentOnceMap.get(groupValue) == null) {
+                    int numSubQueries = subScoresByQuery.length;
+                    LeafFieldComparator[] comparators = comparatorsMap.get(groupValue);
+                    FieldValueHitQueue<FieldValueHitQueue.Entry>[] compoundScores = groupQueueMap.get(groupValue);
+                    for (int i = 0; i < numSubQueries; i++) {
+                        LeafFieldComparator[] leafFieldComparators = compoundScores[i].getComparators(context);
+                        int[] reverseMuls = compoundScores[i].getReverseMul();
+                        if (leafFieldComparators.length == 1) {
+                            reverseMulMap.put(groupSelector.copyValue(), reverseMuls[0]);
+                            comparators[i] = leafFieldComparators[0];
+                        } else {
+                            reverseMulMap.put(groupSelector.copyValue(), 1);
+                            comparators[i] = new MultiLeafFieldComparator(leafFieldComparators, reverseMuls);
+                        }
+                        comparators[i].setScorer(compoundQueryScorer);
+                    }
+                    comparatorsMap.put(groupSelector.copyValue(), comparators);
+                    initializeLeafComparatorsPerSegmentOnceMap.put(groupSelector.copyValue(), false);
+                }
+                totalHitCount++;
+                hitsThresholdChecker.incrementHitCount();
+
+                if (hitsThresholdChecker.isThresholdReached()) {
+                    setTotalHitsRelation(TotalHits.Relation.GREATER_THAN_OR_EQUAL_TO);
+                    log.info("Terminating collection as hits threshold is reached");
+                    throw new CollectionTerminatedException();
```

Comment:
when this happen, if that's user facing exception we need to log with error level and probably add more details about this error?

Reply:
Sure, I can add more information.  This was taken directly from the sort collector.